src/anim/evaluator/anim-target-value.js

```diff
@@ -58,17 +85,37 @@ class AnimTargetValue {
     updateValue(index, value) {
         // always reset the value of the target when the counter is 0
         if (this.counter === 0) {
-            this.value[0] = 0;
-            this.value[1] = 0;
-            this.value[2] = 0;
-            this.value[3] = 1;
+            AnimEvaluator._set(this.value, [0, 0, 0, 1], this.valueType);
+            if (!this._normalizeWeights) {
+                AnimEvaluator._blend(this.value, this.baseValue, 1, this.valueType);
+            }
         }
         if (!this.mask[index]) return;
-        if (this.counter === 0) {
-            AnimEvaluator._set(this.value, value, this.valueType);
+        if (this._layerBlendType(index) === ANIM_LAYER_ADDITIVE && !this._normalizeWeights) {
```

Comment:
this whole block of code is performance critical path, it'd be good to make it very optimized, it does not seem to be at the moment. Even `...` likely allocates array under the hood.

Reply:
I've removed the spread operations here, it's now just setting individual values from the various arrays.